Userland/Services/RequestServer/ConnectionCache.h

```diff
@@ -185,84 +188,86 @@ decltype(auto) get_or_create_connection(auto& cache, URL::URL const& url, auto j
     // issues with concurrent connections, so we'll only allow one connection per URL in that case to avoid issues.
     // This is a bit too aggressive, but there's no way to know if the server can handle concurrent connections
     // without trying it out first, and that's not worth the effort as HTTP/1.0 is a legacy protocol anyway.
-    auto it = sockets_for_url.find_if([&](auto& connection) { return properties.requests_served_per_connection < 2 || connection->request_queue.is_empty(); });
-    auto did_add_new_connection = false;
-    auto failed_to_find_a_socket = it.is_end();
-    if (failed_to_find_a_socket && sockets_for_url.size() < ConnectionCache::MaxConcurrentConnectionsPerURL) {
-        using ConnectionType = RemoveCVReference<decltype(*cache.begin()->value->at(0))>;
-        auto connection_result = proxy.tunnel<typename ConnectionType::SocketType, typename ConnectionType::StorageType>(url);
-        if (connection_result.is_error()) {
-            dbgln("ConnectionCache: Connection to {} failed: {}", url, connection_result.error());
-            Core::deferred_invoke([job] {
-                job->fail(Core::NetworkJob::Error::ConnectionFailed);
-            });
-            return ReturnType { nullptr };
+    return cache.with_locked([&] (auto&) -> ReturnType {
```

Comment:
this is an absolutely massive lambda. Can we yoink its internals into a function that verifies that the cache is locked?

Reply:
...how do you verify anything with a MutexProtected?
will move it out into a different function tho

Reply:
Moved to a new function.